src/query/storages/fuse/src/operations/read_data.rs

```diff
@@ -73,6 +77,42 @@ impl FuseTable {
         }
     }
 
+    fn apply_data_mask_policy_if_needed(
+        &self,
+        ctx: Arc<dyn TableContext>,
+        plan: &DataSourcePlan,
+        pipeline: &mut Pipeline,
+    ) -> Result<()> {
+        if let Some(data_mask_policy) = &plan.data_mask_policy {
+            let num_input_columns = plan.schema().num_fields();
+            let mut exprs = Vec::with_capacity(num_input_columns);
+            for i in 0..num_input_columns {
+                let expr = data_mask_policy
+                    .get(&i)
+                    .map(|remote_expr| remote_expr.as_expr(&BUILTIN_FUNCTIONS));
+
+                exprs.push(expr);
+            }
+
+            let op = BlockOperator::Replace { exprs };
```

Comment:
There is no need for an additional `BlockOperator::Replace`; a `BlockOperator::Map` will perfectly do the job

Reply:
ok, use `Map` and `Project` instead